### PR TITLE
Issue1174 improved callbacks

### PR DESCRIPTION
**Pull request by MatthiasJE**
This is the pull request for #1174. It does exactly what I proposed in that issue.

### CODE REVIEW

File: odl/solvers/util/callback.py [hunk 1]
```diff
@@ -445,11 +457,64 @@ def __repr__(self):
         return '{}({})'.format(self.__class__.__name__, inner_str)
 
 
+class CallbackPrintTotalTime(Callback):
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: Why do we need a new class for this? 
IMO we should instead just add a `cumulative=False` option to the existing `CallbackPrintTiming` and massage the code a bit to cover both cases.

Comment (adler-j) [reply odl/solvers/util/callback.py]: Actually this is a great suggestion. Perhaps that should be done instead.

Comment (MatthiasJE) [reply odl/solvers/util/callback.py]: Agree. I will make those changes.

File: odl/solvers/util/callback.py [hunk 2]
```diff
@@ -416,25 +423,37 @@ def __init__(self, fmt='Time elapsed = {:<5.03f} s', step=1):
...
+        self.time = 0
+        self.kwargs = kwargs
 
     def __call__(self, _):
         """Print time elapsed from the previous iteration."""
         if self.iter % self.step == 0:
             t = time.time()
-            print(self.fmt.format(t - self.time))
-            self.time = t
+            self.time = t - self.base_time
```
Comment (adler-j) [inline odl/solvers/util/callback.py]: Is there any reason why we need to save this as `self...`?

Comment (MatthiasJE) [reply odl/solvers/util/callback.py]: I was thinking that it was good to have the cumulative time as an output attribute but actually it is cleaner for this not to be. This is now fixed.

**Pull request merged**